homeassistant/components/calendar/todoist.py

```diff
@@ -116,7 +122,8 @@
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities,
```

Comment:
the setup method only does I/O and so should not be async. It's inefficient to call executor a lot.

You can still define async methods inside the setup platform.